src/headers/utils.h

```diff
@@ -8,21 +8,23 @@
 
 using namespace std;
 
+vector<vector<double>> init_matrix(int rows, int cols);
+
 class Matrix
 {
 private:
     size_t cols;
     size_t rows;
-    double **matrix;
+    //double **matrix;
     std::vector<std::vector<double>> m;
```

Comment:
I think we should use more than one character as the variable name.
```suggestion
    std::vector<std::vector<double>> matrix;
```